### PR TITLE
feat(server): add database schema models

### LINKED ISSUE #23: Create Database Models in server folder
We need to create data models in MongoDB. Follow the models mentioned [here](https://github.com/pesto-students/batch-9-Lumen/wiki/Architecture-for-the-product).

### PR DESCRIPTION
**Pull request by karansapolia**
fixes: #23

## PR Checklist
Please check that your PR fulfills the following requirements:

- [x] The commit message follows [our guidelines](https://github.com/pesto-students/batch-9-Lumen/wiki/Commit-message-convention)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] GIFs/Videos/Images added (only for frontend changes)

## PR Type
What kind of change does this PR introduce? 

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Infrastructure changes
- [ ] Other... Please describe:
Issue Number: 
#

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: server/models/blogs.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    ref: 'User',
+  },
+  title: {
+    type: String,
+    required: true,
+  },
+  content: [
+    {
+      _id: Schema.Types.ObjectId,
+      type: Schema.Types.Mixed,
```
Comment (nihalanisumit) [inline server/models/blogs.js]: Why two types?

**Review by nihalanisumit — APPROVED** [review]
LGTM!

File: server/models/blogs.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      content: {
+        type: Schema.Types.Mixed,
+        required: true,
+      },
+    },
```
Comment (piyushagwl007) [inline server/models/blogs.js]: `Content` has a child as `content` which seems confusing.

File: server/models/blogs.js (hunk 1 above)
Comment (karansapolia) [reply server/models/blogs.js]: Right! So type should be a string describing the type of the inner content here. I'll update that

File: server/models/blogs.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  },
+  title: {
+    type: String,
+    required: true,
+  },
+  content: [
+    {
+      _id: Schema.Types.ObjectId,
+      type: Schema.Types.Mixed,
+      content: {
```
Comment (nimish-gupta) [inline server/models/blogs.js]: Content within content? A better key nMe can be used here. Also why not create another content collection and then use ref ?

File: server/models/blogs.js (hunk 2 above)
Comment (piyushagwl007) [reply server/models/blogs.js]: Also, I think this is not the appropriate use of Mixed type schema

File: server/models/user.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import mongoose from 'mongoose';
+
+const { Schema } = mongoose;
+
+const UserSchema = new Schema({
```
Comment (nimish-gupta) [inline server/models/user.js]: Timestamp?

File: server/models/blogs.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  },
+  title: {
+    type: String,
+    required: true,
+  },
+  content: [
+    {
+      _id: Schema.Types.ObjectId,
+      type: String,
+      contentFragment: {
```
Comment (nimish-gupta) [inline server/models/blogs.js]: Why not separate collection and then joined by tef?

File: server/models/blogs.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import mongoose from 'mongoose';
+
+const { Schema } = mongoose;
+
+const BlogsSchema = new Schema({
```
Comment (nimish-gupta) [inline server/models/blogs.js]: Timestamps?

File: server/models/blogs.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    type: [Schema.Types.ObjectId],
+    ref: 'User',
+  },
+  private: {
+    type: Boolean,
+  },
+});
+
+const Blogs = mongoose.model('Blogs', BlogsSchema);
+module.exports = Blogs;
```
Comment (piyushagwl007) [inline server/models/blogs.js]: export default should be used. We are following the ES6 format.

**Review by nimish-gupta — CHANGES_REQUESTED** [review]

File: server/models/user.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+    required: true,
+  },
+  username: {
+    type: String,
+    unique: true,
+  },
+});
+
+const User = mongoose.model('User', UserSchema);
+module.exports = User;
```
Comment (piyushagwl007) [inline server/models/user.js]: Newline missing at end of the file and here again use export default

**Review by piyushagwl007 — CHANGES_REQUESTED** [review]
Please look into the requested changes.

File: server/models/blogs.js (hunk 7 above)
Comment (karansapolia) [reply server/models/blogs.js]: Sure.

File: server/models/user.js (hunk 8 above)
Comment (karansapolia) [reply server/models/user.js]: :+1:

File: server/models/blogs.js (hunk 2 above)
Comment (karansapolia) [reply server/models/blogs.js]: Both issues addressed and resolved

File: server/models/blogs.js (hunk 3 above)
Comment (karansapolia) [reply server/models/blogs.js]: 1. Better name assigned to inner `content`.
2. The `contentFragment` object represents tagged fragments of a blog post. For example, each paragraph is a different fragment, each image will be its own fragment, similarly for each video file.This is being done to make displaying rich media content in the blog posts easier. 
In this respect, having the contentFragment object as a document outside the `content` array or the BlogSchema document does not make a lot of sense.

Thoughts?

File: server/models/blogs.js (hunk 5 above)
Comment (karansapolia) [reply server/models/blogs.js]: [The same reasoning as before](https://github.com/pesto-students/batch-9-Lumen/pull/43#discussion_r314897577)
Thoughts?

Comment (piyushagwl007) [reply server/models/blogs.js]: You can just do a `fragment` instead of `contentFragment`

**Review by piyushagwl007 — APPROVED** [review]
Just do the changes requested otherwise LGTM.

File: server/models/blogs.js (hunk 6 above)
Comment (karansapolia) [reply server/models/blogs.js]: Right! That would be useful here. Including it

File: server/models/user.js (hunk 4 above)
Comment (karansapolia) [reply server/models/user.js]: Adding it. Could be useful.

**Review by nimish-gupta — APPROVED** [review]
LGTM <3

> Dont know heart emoji short form :(

**Pull request merged**